src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -2357,3 +2366,1721 @@ IsForeignTable(Oid relationId)
+						   indexOK, NULL, lengthof(scanKey), scanKey);
+
+	JobTaskStatusCounts counts = { 0 };
+	HeapTuple heapTuple = NULL;
+	while (HeapTupleIsValid(heapTuple = systable_getnext(scanDescriptor)))
+	{
+		Datum values[Natts_pg_dist_background_task] = { 0 };
+		bool isnull[Natts_pg_dist_background_task] = { 0 };
+
+		heap_deform_tuple(heapTuple, tupleDescriptor, values, isnull);
+
+		Oid statusOid = DatumGetObjectId(values[Anum_pg_dist_background_task_status -
+												1]);
+		BackgroundTaskStatus status = BackgroundTaskStatusByOid(statusOid);
+
+		switch (status)
+		{
+			case BACKGROUND_TASK_STATUS_BLOCKED:
+			{
+				counts.blocked++;
+				break;
+			}
+
+			case BACKGROUND_TASK_STATUS_RUNNABLE:
+			{
+				counts.runnable++;
+				break;
+			}
+
+			case BACKGROUND_TASK_STATUS_RUNNING:
+			{
+				counts.running++;
+				break;
+			}
+
+			case BACKGROUND_TASK_STATUS_DONE:
+			{
+				counts.done++;
+				break;
+			}
+
+			case BACKGROUND_TASK_STATUS_ERROR:
+			{
+				counts.error++;
+				break;
+			}
+
+			case BACKGROUND_TASK_STATUS_UNSCHEDULED:
+			{
+				counts.unscheduled++;
+				break;
+			}
+
+			case BACKGROUND_TASK_STATUS_CANCELLED:
+			{
+				counts.cancelled++;
+				break;
+			}
+
+			case BACKGROUND_TASK_STATUS_CANCELLING:
+			{
+				counts.cancelling++;
+				break;
+			}
+
+			default:
+			{
+				elog(ERROR, "unknown state in pg_dist_background_task");
+			}
+		}
+	}
+
+	systable_endscan(scanDescriptor);
+	table_close(pgDistBackgroundTasks, NoLock);
+
+	return counts;
+}
+
+
+/*
+ * SetFieldValue populates values, isnull, replace according to the newValue passed,
+ * returning if the value has been updated or not.
+ *
+ * suggested use would be:
+ *   updated |= SetFieldValue(Anum_...._...., isnull, replace, values, newValue);
+ *   updated |= SetFieldNull(Anum_...._...., isnull, replace, values);
+ *
+ * Only if updated is set in the end the tuple has to be updated in the catalog.
+ */
+static bool
+SetFieldValue(int attno, Datum values[], bool isnull[], bool replace[], Datum newValue)
+{
+	int idx = attno - 1;
+	bool updated = false;
+
+	if (isnull[idx])
+	{
+		isnull[idx] = false;
+		updated = true;
+	}
+
+	if (newValue != values[idx])
+	{
+		values[idx] = newValue;
+		updated = true;
+	}
+
+	replace[idx] = updated;
+	return updated;
+}
+
+
+/*
+ * SetFieldNull populates values, isnull and replace according to a null value,
+ * returning if the value has been updated or not.
+ *
+ * suggested use would be:
+ *   updated |= SetFieldValue(Anum_...._...., isnull, replace, values, newValue);
+ *   updated |= SetFieldNull(Anum_...._...., isnull, replace, values);
+ *
+ * Only if updated is set in the end the tuple has to be updated in the catalog.
+ */
+static bool
+SetFieldNull(int attno, Datum values[], bool isnull[], bool replace[])
+{
+	int idx = attno - 1;
+	bool updated = false;
+
+	if (!isnull[idx])
+	{
+		isnull[idx] = true;
+		updated = true;
+	}
+	values[idx] = InvalidOid;
+
+	replace[idx] = updated;
+	return updated;
```

Comment:
nit: I think readability of this function improves if you use an early return

```suggestion
	int idx = attno - 1;
	if (isnull[idx])
	{
		return false;
	}
	isnull[idx] = true;
	values[idx] = InvalidOid;
	replace[idx] = true;
	return true;
```